glusterd2/commands/volumes/volume-reset.go

```diff
@@ -89,17 +90,32 @@ func volumeResetHandler(w http.ResponseWriter, r *http.Request) {
 		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
 		return
 	}
-
+	//save volume information for transaction failure scenario
+	if err := txn.Ctx.Set("oldvolinfo", oldvolinfo); err != nil {
+		logger.WithError(err).Error("failed to set oldvolinfo in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
 	txn.Steps = []*transaction.Step{
 		{
-			DoFunc: "vol-option.UpdateVolinfo",
-			Nodes:  []uuid.UUID{gdctx.MyUUID},
+			DoFunc:   "vol-option.XlatorActionDoReSet",
+			UndoFunc: "vol-option.XlatorActionUndoReSet",
+			Nodes:    volinfo.Nodes(),
```

Comment:
This also needs the skip optimization:

```go
Skip:     !isActionStepRequired(&req)
```

You can move implementation of `isActionStepRequired(&req)` to `glusterd2/commands/volumes/common.go`

You will have to make  `isActionStepRequired` take a generic map. That will allow you to reuse it.